scripts/docs/generate-transformations.test.ts

```diff
@@ -0,0 +1,56 @@
+import { getMarkdownContent, getJavaScriptContent, readMeContent } from './generate-transformations.ts';
+
+describe('makefile script tests', () => {
+  it('should execute without error and match the content written to index.md', () => {
+    // Normalize and compare.
+    expect(contentDoesMatch(getJavaScriptContent(), getMarkdownContent())).toBe(true);
+  });
+
+  it('should be able to tell if the content DOES NOT match', () => {
+    const wrongContent = getJavaScriptContent().concat('additional content to mismatch');
+    // Normalize and compare.
+    expect(contentDoesMatch(wrongContent, getMarkdownContent())).toBe(false);
+
+    // If this test fails, refer to `./docs/README.md` "Content guidelines" for more information
+    // about editing and building the Transformations docs.
+  });
+});
+
+export function contentDoesMatch(jsContent: string, mdContent: string): Boolean {
+  return normalizeContent(jsContent) === normalizeContent(mdContent);
+}
+
+/* 
+  Normalize content by removing all whitespace (spaces, tabs, newlines, carriage returns, 
+  form feeds, and vertical tabs) and special characters.
+
+  NOTE: There are numerous unpredictable formatting oddities when pasring javascript to markdown;
+  almost all of them are irrelevant to the actual content of the file, which is why we strip them out here.
```

Comment:
One typo and some nits.

```suggestion
  NOTE: There are numerous unpredictable formatting oddities when transforming JavaScript to Markdown;
  almost all of them are irrelevant to the actual content of the file, which is why we strip them out here.
```

On the technicalities of this statement, I think most (probably all) Markdown parsers would correctly parse the JavaScript table as it is. I think it is a `prettier` formatting change that results in the Markdown table differences.

That being said, I think it's worth using the `prettier` formatted Markdown for a few reasons:
1. It's simpler for tools to roundtrip modifications. 
1. It helps alleviate the ambiguities in the Markdown specs.
1. Consistent formatting is easier for humans to read and parse.

I guess what we are actually talking about is parsing differences between `prettier` and whatever parsing library is used by the Grafana UI. I'm not sure how important that distinction is in terms of function and variable naming, just thought it's worth mentioning :)